mlir/lib/Analysis/FlatLinearValueConstraints.cpp

```diff
@@ -86,7 +86,8 @@ getFlattenedAffineExprs(ArrayRef<AffineExpr> exprs, unsigned numDims,
     if (!expr.isPureAffine())
       return failure();
 
-    flattener.walkPostOrder(expr);
+    auto flattenResult = flattener.walkPostOrder(expr);
+    assert(succeeded(flattenResult) && "affine expr containts poison expr");
```

Comment:
Why does this warrant an assertion? Could we choose to materialize a different constant?